shared/js/client_api/channels.js

```diff
@@ -22,9 +22,9 @@ module.exports = {
       contentType: "application/json; charset=utf-8",
       data: JSON.stringify(value)
     }).done(function(data, text) {
-      callback(data);
+      callback(null, data);
     }).fail(function(request, status, error) {
-      callback(undefined);
+      callback(error, undefined);
```

Comment:
Really I think our callback should probably be always returning null for the data portion when there's an error, but I'm not sure if existing callers depending on seeing certain falsey values, so I'm leaving this as is for now.